pjsip/src/pjsip/sip_parser.c

```diff
@@ -220,7 +220,9 @@ PJ_INLINE(void) parser_get_and_unescape(pj_scanner *scanner, pj_pool_t *pool,
 #else
     PJ_UNUSED_ARG(unesc_spec);
     pj_scan_get(scanner, spec, token);
-    *token = pj_str_unescape(pool, token);
+    if (!pj_cis_match(spec, '%')) {
+        *token = pj_str_unescape(pool, token);
+    }
```

Comment:
From the CI, looks like this fails pjsip unit test?
This function is used in many places so changing it directly does sound quite risky.

An alternative approach is to modify `int_parse_via_param()` implementation: parse the param value based on the param name, for example, param `branch` should use simply `pj_scan_get()` instead of `parser_get_and_unescape()`, while other params (including `other-param`) can use `parser_get_and_unescape()`. Need to be careful with quoted string (which is currently already taken care by `parse_param_imp()`.

Or do the param name check in the `parse_param_imp()`, which already handles quoted string, perhaps by introducing a new option such as `PJSIP_PARSE_VIA_MODE`, so when that option is set, [this block](https://github.com/pjsip/pjproject/blob/master/pjsip/src/pjsip/sip_parser.c#L1243) will not escape the param value if the param name is `branch`.